addon/modifiers/did-intersect.js

```diff
@@ -38,15 +68,64 @@ export default class DidIntersectModifier extends Modifier {
       return;
     }
 
-    let { onEnter, onExit, options } = this.args.named;
+    let { onEnter, onExit, maxEnter, maxExit, options } = this.args.named;
 
     assert('onEnter or/and onExit is required', onEnter || onExit);
 
-    if (onEnter) {
-      this.observerManager.addEnterCallback(this.element, onEnter);
+    // We should technically accept 0, in case of programmatic changes
+    if (maxEnter || maxEnter === 0) {
+      assert('maxEnter must be an integer', Number.isInteger(maxEnter));
+      this._maxEnterIntersections = maxEnter;
+    }
+
+    // We should accept 0 here too.
+    if (maxExit || maxExit === 0) {
+      assert('maxExit must be an integer', Number.isInteger(maxExit));
+      this._maxExitIntersections = maxExit;
+    }
+
+    if (onEnter && !this._isExceedingMaxEnters) {
+      this.observerManager.addEnterCallback(this.element, (...args) => {
+        if (this.isDestroying) {
+          return;
+        }
+
+        this._numOfEnters++;
+
+        if (
+          this._isExceedingMaxEnters &&
+          (!onExit || this._isExceedingMaxExits)
```

Comment:
We should save `onEnter` and `onExit` as properties of the modifier instead of referencing them by scope like this. If these callbacks change, it's possible that this scoping would actually execute the old callback methods. By doing this, we also should endeavor to avoid calling `addEnterCallback` or `addExitCallback` more than once. That is, we should track if we have already set up the callbacks and skip setting them up again in line 87 and 109 if they already exist (and ensure we reset that tracking in `unobserve`. Otherwise, we could in theory set up multiple callbacks, since the observer manager technically allows the element to be observed an arbitrary number of times, yes?